pyvista/core/filters/image_data.py

```diff
@@ -986,6 +1014,578 @@ def contour_labeled(
+
+            Since the smoothing operation occurs across all input regions, using this
+            option to filter the output means that the selected output regions will have
+            the same shape (i.e. smoothed in the same manner), regardless of the outputs
+            that are selected. This is useful for generating a surface for specific
+            labels while also preserving sharp boundaries with non-selected outputs.
+
+            .. note::
+
+                Selecting outputs does not affect whether a boundary polygon is
+                considered to be ``internal`` or ``external``. That is, an internal
+                boundary remains internal even if only one of the two foreground regions
+                on the boundary is selected.
+
+        closed_surface : bool, default: True
+            Generate polygons to "close" the surface at the boundaries of the image.
+            This option is only relevant when there are foreground regions on the border
+            of the image. Setting this value to ``False`` is useful if processing multiple
+            volumes separately so that the generated surfaces fit together without
+            creating surface overlap.
+
+        output_mesh_type : str, default: None
+            Type of the output mesh. Can be either ``'quads'``, or ``'triangles'``. By
+            default, the output mesh has :attr:`~pyvista.CellType.TRIANGLE` cells when
+            ``smoothing`` is enabled and :attr:`~pyvista.CellType.QUAD` cells (quads)
+            otherwise. The mesh type can be forced to be triangles or quads; however,
+            if smoothing is enabled and the type is ``'quads'``, the generated quads
+            may not be planar.
+
+        scalars : str, optional
+            Name of scalars to process. Defaults to currently active scalars. If cell
+            scalars are specified, the input image is first re-meshed with
+            :meth:`~pyvista.ImageDataFilters.cells_to_points` to transform the cell
+            data into point data.
+
+        compute_normals : bool, default: True
+            Compute point and cell normals for the contoured output using
+            :meth:`~pyvista.PolyDataFilters.compute_normals` with ``auto_orient_normals``
+            enabled by default. If ``False``, the generated polygons may have
+            inconsistent ordering and orientation (and may negatively impact
+            the shading used for rendering).
+
+            .. warning::
+
+                Enabling this option is likely to generate surfaces with normals
+                pointing outward when ``closed_surface`` is ``True`` and
+                ``boundary_style`` is ``True`` (the default). However, this is
+                not guaranteed if the generated surface is not closed or if internal
+                boundaries are generated. Do not assume the normals will point outward
+                in all cases.
+
+        multi_component_output : bool, optional
```

Comment:
Should this be named `multi_component_scalars` instead? Or does this imply a scalar name such that `str` is expected (scalar name) instead of `bool` type?

```suggestion
        multi_component_output : bool, optional
```